### PR TITLE
 Start setting up a pilot simulation driven by SOSE part 2

### PR DESCRIPTION
**Pull request by ali-ramadhan**

### CODE REVIEW

File: src/pilot_simulation.jl [hunk 1]
```diff
@@ -123,23 +130,214 @@ forcings = ModelForcing(u=Fu′, v=Fv′, w=Fw′, T=Fθ′, S=Fs′)
...
+#####
+##### Initial conditions
+#####
+
+ε(μ) = μ * randn() # noise
+
+U₀(x, y, z) = ℑU(0, z)
+V₀(x, y, z) = ℑV(0, z)
+W₀(x, y, z) = ε(1e-10)
+Θ₀(x, y, z) = ℑΘ(0, z)
```
Comment (glwagner) [inline src/pilot_simulation.jl]: Is "0" for "time"? Should we use the function signature `z, t`?

File: src/pilot_simulation.jl [hunk 2]
```diff
@@ -123,23 +130,214 @@ forcings = ModelForcing(u=Fu′, v=Fv′, w=Fw′, T=Fθ′, S=Fs′)
...
+
+#####
+##### Initial conditions
+#####
+
+ε(μ) = μ * randn() # noise
+
+U₀(x, y, z) = ℑU(0, z)
+V₀(x, y, z) = ℑV(0, z)
+W₀(x, y, z) = ε(1e-10)
```
Comment (glwagner) [inline src/pilot_simulation.jl]: Could make this dimensional by setting to some small fraction of some measure of the horizontal velocities

File: src/pilot_simulation.jl [hunk 3]
```diff
@@ -123,23 +130,214 @@ forcings = ModelForcing(u=Fu′, v=Fv′, w=Fw′, T=Fθ′, S=Fs′)
...
+Θ₀(x, y, z) = ℑΘ(0, z)
+S₀(x, y, z) = ℑS(0, z)
+
+Oceananigans.set!(model, u=U₀, v=V₀, w=W₀, T=Θ₀, S=S₀)
+
+#####
+##### Setting up diagnostics
+#####
```
Comment (glwagner) [inline src/pilot_simulation.jl]: We should probably set up utilities that define long lists of diagnostics

File: src/pilot_simulation.jl [hunk 4]
```diff
@@ -123,23 +130,214 @@ forcings = ModelForcing(u=Fu′, v=Fv′, w=Fw′, T=Fθ′, S=Fs′)
...
+profile_dims = Dict(k => ("zC",) for k in keys(profiles))
+profile_dims["w"] = ("zF",)
+
+profile_output_writer =
+    NetCDFOutputWriter(model, profiles, filename=filename_prefix * "_profiles.nc", interval=10minute,
+                      global_attributes=global_attributes, output_attributes=output_attributes,
+                      dimensions=profile_dims)
+
+#####
+##### Large scale solution output writer
```
Comment (glwagner) [inline src/pilot_simulation.jl]: Large scale = 3D fields?

File: src/pilot_simulation.jl [hunk 5]
```diff
@@ -70,10 +75,12 @@ S = reverse(S, dims=2)
...
 #####
 
-Nx = Ny = Nz = 32
-Lx = Ly = Lz = 100
-topology = (Periodic, Bounded, Bounded)
-grid = RegularCartesianGrid(topology=topology, size=(Nx, Ny, Nz) x=(0, Lx), y=(0, Ly), z=(-Lz, 0))
+Nx = Ny = 32
+Nz = 2Nx
+Lx = Ly = 1000.0
+Lz = 2Lx
```
Comment (glwagner) [inline src/pilot_simulation.jl]: Should we analyze the SOSE data and set the domain depth to 2x and the domain width to 4x the maximum boundary layer depth observed in the data?

File: src/pilot_simulation.jl (hunk 1 above)
Comment (ali-ramadhan) [reply src/pilot_simulation.jl]: Yeah `z, t` might be clearer, just haven't flipped it around as xarray returns data with `t, z` (might depend on how NetCDF orders the coordinates but I'm sure it's easy to change).

File: src/pilot_simulation.jl (hunk 2 above)
Comment (ali-ramadhan) [reply src/pilot_simulation.jl]: Ah good idea!

File: src/pilot_simulation.jl (hunk 4 above)
Comment (ali-ramadhan) [reply src/pilot_simulation.jl]: Ah sorry it's the base state so really just vertical profiles as a function of time.

**Pull request merged**